packages/web-components/src/components/carousel/carousel.ts

```diff
@@ -111,6 +114,39 @@ class DDSCarousel extends LitElement {
     }
   }
 
+  /**
+   * Handles card focus throughout pages.
+   *
+   * @param event The event.
+   */
+  @HostListener('shadowRoot:focusin')
+  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
+  private _handleFocus = async ({ target, relatedTarget }: FocusEvent) => {
+    const currentContains = target !== this && this.contains(target as DDSCard);
+    const oldContains = relatedTarget !== this && this.contains(relatedTarget as DDSCard);
+
+    const currentChildIndex = Array.from(this.children).indexOf(target as HTMLElement);
+    const oldChildIndex = Array.from(this.children).indexOf(relatedTarget as HTMLElement);
+
+    const targetPageIndex = currentChildIndex % this.pageSize;
+
+    // reset to first page if tabbing back into the carousel after previously moving pages
+    if (currentContains && currentChildIndex === 0) {
+      this.start = 0;
+    }
+
+    if (oldContains && currentContains && currentChildIndex < this._total) {
+      // going forwards, change page if focused card is the first on the next page
+      if (oldChildIndex < currentChildIndex && currentChildIndex !== 0 && targetPageIndex === 0) {
+        this._handleClickNextButton();
+
+        // going backwards, change page if focused card is the last on the previous page
+      } else if (currentChildIndex < oldChildIndex && targetPageIndex === this.pageSize - 1) {
+        this._handleClickPrevButton();
+      }
+    }
+  };
```

Comment:
I expect the logic here to be simply updating `start` if the newly focused card is out of the page range, instead of detecting "hit the boundary" and using the event handler for prev/next button. Would you be able to try?

Reply:
That's what it does; if the newly focused card is out of range, it goes to the next/prev page. The handlers are doing the `start` updating logic themselves, so I think it'd be redundant updating `start` on its own.

Reply:
Wait nevermind, I think I see what you mean, let me give it a try